### PR TITLE
hot fix: fix unban all peers and replace transactions from pool to queue

### PR DESCRIPTION
**Pull request by bogdan34**

### CODE REVIEW

File: core/service/sync.ts [hunk 1]
```diff
@@ -55,11 +55,16 @@ export class SyncService implements ISyncService {
         const errors: Array<string> = [];
         const filteredPeers = PeerRepository.getPeersByFilter(lastBlock.height, SystemRepository.broadhash);
 
-        if (!filteredPeers.length) {
+        const unbannedPeers = filteredPeers.filter(peer => !PeerRepository.isBanned(peer));
+
+        if (unbannedPeers.length === 0 && filteredPeers.length !== 0) {
+            return new ResponseEntity({ errors: [ERROR_ALL_PEERS_ARE_BANNED] });
+        }
+        if (!unbannedPeers.length) {
```
Comment (macaqueee) [inline core/service/sync.ts]: move it above

File: shared/model/round.ts [hunk 2]
```diff
@@ -26,4 +26,19 @@ export class Round extends RoundModel {
         return new Round(this);
     }
 
+    static compareRounds(r1: Round, r2: Round): boolean {
```
Comment (Verbalist) [inline shared/model/round.ts]: util

File: core/service/round.ts [hunk 3]
```diff
@@ -145,6 +145,21 @@ class RoundService implements IRoundService {
...
+    public regenerateRound() {
+        const prevRound = RoundRepository.getPrevRound();
+        if (!prevRound) {
+            return;
+        }
+        const newRound = this.generate(getLastSlotNumberInRound(prevRound) + 1);
+        const currentRound = RoundRepository.getCurrentRound();
+
+        if (!compareRounds(currentRound, newRound)) {
+            RoundRepository.deleteLastRound();
```
Comment (axelsheva) [inline core/service/round.ts]: Where are deleting blocks of a current round?

File: core/service/transaction.ts [hunk 4]
```diff
@@ -477,6 +477,13 @@ class TransactionService<T extends IAsset> implements ITransactionService<T> {
     popFromPool(limit: number): Array<Transaction<IAsset>> {
         return TransactionPool.popSortedUnconfirmedTransactions(limit);
     }
+
+    returnToQueueAllTransactionFromPool(): void {
+        TransactionPool.popSortedUnconfirmedTransactions(TransactionPool.getSize())
+            .forEach(transaction => {
+                TransactionQueue.push(transaction);
```
Comment (axelsheva) [inline core/service/transaction.ts]: These transactions will be immediately processed?

File: core/service/round.ts [hunk 5]
```diff
@@ -183,7 +198,7 @@ class RoundService implements IRoundService {
                 }
                 this.backwardProcess();
             }
-
+            //  TODO если откатываемся к генезис блоку, оно зацыкливается, нужно пофиксить
```
Comment (axelsheva) [inline core/service/round.ts]: Russian language

File: core/repository/sync.ts [hunk 6]
```diff
@@ -19,8 +20,8 @@ interface ISyncRepo {
 
     sendUnconfirmedTransaction(trs: Transaction<any>): void;
 
-    requestCommonBlocks(blockData: { id: string, height: number }):
-        Promise<ResponseEntity<{ isExist: boolean, peer: Peer }>>;
+    requestCommonBlocks(blockData: { id: string, height: number }, peer: Peer):
```
Comment (macaqueee) [inline core/repository/sync.ts]: create `type` for that

File: core/util/round.ts [hunk 7]
```diff
@@ -2,3 +2,17 @@ import { Round, Slot } from 'shared/model/round';
...
+
+export const compareRounds = (r1: Round, r2: Round): boolean => {
+    let flag = true;
+    if (Object.keys(r1).length !== Object.keys(r2).length) {
+        return false;
+    }
+    Object.keys(r2.slots).forEach((generatorPublicKey: string) => {
+        if (!(generatorPublicKey in r1.slots) ||
+            r2.slots[generatorPublicKey].slot !== r1.slots[generatorPublicKey].slot) {
+            flag = false;
```
Comment (axelsheva) [inline core/util/round.ts]: and break?

**Pull request closed without merging**